### PR TITLE
Add `RemovedComponentEvents::iter`

### PR DESCRIPTION
**Pull request by Shatur**
# Objective

Sometimes it's useful to iterate over removed entities. For example, in my library [bevy_replicon](https://github.com/projectharmonia/bevy_replicon) I need it to iterate over all removals to replicate them over the network.

Right now we do lookups, but it would be more convenient and faster to just iterate over all removals.

## Solution

Add `RemovedComponentEvents::iter`.

---

### Added

- `RemovedComponentEvents::iter` to iterate over all removed components.

### CODE REVIEW

File: crates/bevy_ecs/src/removal_detection.rs [hunk 1]
```diff
@@ -83,6 +83,11 @@ impl RemovedComponentEvents {
         }
     }
 
+    /// Returns iterator over components and their entity events.
```
Comment (pablo-lua) [inline crates/bevy_ecs/src/removal_detection.rs:86]:
```suggestion
    /// Returns iterator an over components and their entity events.
```

**Review by pablo-lua — APPROVED** [review]
Looks good and simple. Just a typo here

File: crates/bevy_ecs/src/removal_detection.rs (hunk 1 above)
Comment (Shatur) [reply crates/bevy_ecs/src/removal_detection.rs:86]: Maybe you wanted to write `an iterator`?

Comment (pablo-lua) [reply crates/bevy_ecs/src/removal_detection.rs:86]: Yep, top 10 epic fails

**Pull request merged**